datarobot_provider/operators/model_registry.py

```diff
@@ -0,0 +1,97 @@
+from enum import Enum
+from typing import Any
+from typing import Dict
+from typing import Sequence
+
+import datarobot as dr
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.utils.context import Context
+
+from datarobot_provider.hooks.datarobot import DataRobotHook
+
+
+class ModelType(Enum):
+    LEADERBOARD = "leaderboard"
+    CUSTOM = "custom"
+    EXTERNAL = "external"
+
+
+class CreateRegisteredModelVersionOperator(BaseOperator):
```

Comment:
I think it could be good to inherit from `BaseDatarobotOperator`

There's an in-flight PR that's updating existing operators to use this new class:
https://github.com/datarobot/airflow-provider-datarobot/pull/135